sklearn/preprocessing/_discretization.py

```diff
@@ -63,11 +63,11 @@ class KBinsDiscretizer(TransformerMixin, BaseEstimator):
 
     Attributes
     ----------
-    n_bins_ : int array, shape (n_features,)
+    n_bins_ : ndarray of shape (n_features,), dtype=np.int_
```

Comment:
I am wondering why `np.int_` is used here instead of `np.int`.

Reply:
I couldn't find `np.int` to be a valid dtype in https://numpy.org/doc/stable/user/basics.types.html  (similar to the comment in https://github.com/scikit-learn/scikit-learn/pull/18025#issuecomment-667240237). But as I understand it np.int_ is platform dependent and will be either np.int32 or np.int64. Therefore, I thought I could be a good dtype to use here. Please correct me if I'm wrong or misunderstood something. :)

Reply:
I would say `dtype={np.int32, np.int64}` to avoid platform-dependent definitions.

Reply:
`np.int` would be fine with me:

```python
In [1]: import numpy as np

In [2]: np.int
Out[2]: int

In [3]: type(np.int)
Out[3]: type
```

This is a valid type and would be platform-specific.

Reply:
The main issue with np.int is that it's deprecated https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated